Race in deleting golem.db in tests

```
self = <tests.golem.test_client.TestClient testMethod=test_restart_task>
    def tearDown(self):
        if hasattr(self, 'client'):
            self.client.quit()
>       super().tearDown()
tests\golem\test_client.py:88: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
golem\testutils.py:130: in tearDown
    super(DatabaseFixture, self).tearDown()
golem\testutils.py:76: in tearDown
    self.__remove_files()
golem\testutils.py:117: in __remove_files
    shutil.rmtree(self.tempdir)
C:\Python36\lib\shutil.py:494: in rmtree
    return _rmtree_unsafe(path, onerror)
C:\Python36\lib\shutil.py:389: in _rmtree_unsafe
    onerror(os.unlink, fullname, sys.exc_info())
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
path = 'C:\\Users\\buildbot-worker\\AppData\\Local\\Temp\\golem-tests-t3z73szq\\test_restart_task67qo1kfp'
onerror = <function rmtree.<locals>.onerror at 0x0000025A96553488>
    def _rmtree_unsafe(path, onerror):
        try:
            if os.path.islink(path):
                # symlinks to directories are forbidden, see bug #1669
                raise OSError("Cannot call rmtree on a symbolic link")
        except OSError:
            onerror(os.path.islink, path, sys.exc_info())
            # can't continue even if onerror hook returns
            return
        names = []
        try:
            names = os.listdir(path)
        except OSError:
            onerror(os.listdir, path, sys.exc_info())
        for name in names:
            fullname = os.path.join(path, name)
            try:
                mode = os.lstat(fullname).st_mode
            except OSError:
                mode = 0
            if stat.S_ISDIR(mode):
                _rmtree_unsafe(fullname, onerror)
            else:
                try:
>                   os.unlink(fullname)
E                   PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\buildbot-worker\\AppData\\Local\\Temp\\golem-tests-t3z73szq\\test_restart_task67qo1kfp\\golem.db'
```

This might be related:
```
_____ TestKeysAuthWithReactor.test_generate_keys_stop_when_reactor_stopped _____
self = <tests.golem.core.test_keysauth.TestKeysAuthWithReactor testMethod=test_generate_keys_stop_when_reactor_stopped>
logger = <MagicMock name='logger' id='139869773798928'>
    @patch('golem.core.keysauth.logger')
    def test_generate_keys_stop_when_reactor_stopped(self, logger):
        # given
        from twisted.internet import threads
        reactor = self._get_reactor()
    
        # when
        threads.deferToThread(KeysAuth._generate_keys, difficulty=200)
    
        time.sleep(0.01)
        reactor.stop()
        time.sleep(0.01)
    
        # then
        assert not reactor.running
        assert logger.info.call_count == 1
        assert logger.info.call_args_list[0][0][0] == 'Generating new key pair'
>       assert logger.warning.call_count == 1
E       AssertionError: assert 0 == 1
E        +  where 0 = <MagicMock name='logger.warning' id='139869774039976'>.call_count
E        +    where <MagicMock name='logger.warning' id='139869774039976'> = <MagicMock name='logger' id='139869773798928'>.warning
```

--- a/tests/golem/test_client.py
+++ b/tests/golem/test_client.py
@@ -13,8 +13,6 @@ from freezegun import freeze_time
 from pydispatch import dispatcher
 from twisted.internet.defer import Deferred
 
-from golem_sci.implementation import SCIImplementation as sci
-
 import golem
 from apps.appsmanager import AppsManager
 from apps.dummy.task.dummytask import DummyTask
@@ -77,11 +75,10 @@ def done_deferred(return_value=None):
 
 def make_mock_payment_processor(eth=100, gnt=100):
     pp = MagicMock(name="MockPaymentProcessor")
-    pp.ETH_PER_PAYMENT = sci.GAS_PRICE * sci.GAS_PER_PAYMENT
-    pp.ETH_BATCH_PAYMENT_BASE = sci.GAS_PRICE * sci.GAS_BATCH_PAYMENT_BASE
 
-    val = pp.ETH_BATCH_PAYMENT_BASE + pp.ETH_PER_PAYMENT * 10
-    pp.get_gas_cost_per_payment.return_value = val
+    pp.ETH_BATCH_PAYMENT_BASE = 0
+
+    pp.get_gas_cost_per_payment.return_value = 0
 
     pp.gnt_balance.return_value = gnt * denoms.ether, time.time()
     pp.eth_balance.return_value = eth * denoms.ether, time.time()

--- a/tests/golem/test_client.py
+++ b/tests/golem/test_client.py
@@ -13,6 +13,8 @@ from freezegun import freeze_time
 from pydispatch import dispatcher
 from twisted.internet.defer import Deferred
 
+from golem_sci.implementation import SCIImplementation as sci
+
 import golem
 from apps.appsmanager import AppsManager
 from apps.dummy.task.dummytask import DummyTask
@@ -71,6 +73,21 @@ def done_deferred(return_value=None):
     deferred = Deferred()
     deferred.callback(return_value)
     return deferred
+
+
+def make_mock_payment_processor(eth=100, gnt=100):
+    pp = MagicMock(name="MockPaymentProcessor")
+    pp.ETH_PER_PAYMENT = sci.GAS_PRICE * sci.GAS_PER_PAYMENT
+    pp.ETH_BATCH_PAYMENT_BASE = sci.GAS_PRICE * sci.GAS_BATCH_PAYMENT_BASE
+
+    val = pp.ETH_BATCH_PAYMENT_BASE + pp.ETH_PER_PAYMENT * 10
+    pp.get_gas_cost_per_payment.return_value = val
+
+    pp.gnt_balance.return_value = gnt * denoms.ether, time.time()
+    pp.eth_balance.return_value = eth * denoms.ether, time.time()
+    pp._gnt_available.return_value = gnt * denoms.ether
+    pp._eth_available.return_value = eth * denoms.ether
+    return pp
 
 
 @patch(
@@ -537,6 +554,9 @@ class TestClient(TestWithDatabase, TestWithReactor):
     @patch('golem.client.async_run', mock_async_run)
     @patch('golem.network.concent.client.ConcentClientService.start')
     @patch('golem.client.SystemMonitor')
+    @patch('golem.transactions.ethereum.ethereumtransactionsystem.'
+           'PaymentProcessor',
+           return_value=make_mock_payment_processor())
     @patch('golem.client.P2PService.connect_to_network')
     def test_restart_task(self, connect_to_network, *_):
         apps_manager = AppsManager(False)
@@ -553,10 +573,6 @@ class TestClient(TestWithDatabase, TestWithReactor):
             use_docker_manager=False,
             apps_manager=apps_manager
         )
-
-        sci = self.client.transaction_system._sci
-        pp = make_mock_payment_processor(sci)
-        self.client.transaction_system.payment_processor = pp
 
         deferred = Deferred()
         connect_to_network.side_effect = lambda *_: deferred.callback(True)
@@ -899,21 +915,6 @@ class TestTaskCleanerService(TestWithReactor):
         client.clean_old_tasks.assert_called_once()
 
 
-def make_mock_payment_processor(sci, eth=100, gnt=100):
-    pp = MagicMock()
-    pp.ETH_PER_PAYMENT = sci.GAS_PRICE * sci.GAS_PER_PAYMENT
-    pp.ETH_BATCH_PAYMENT_BASE = sci.GAS_PRICE * sci.GAS_BATCH_PAYMENT_BASE
-
-    val = pp.ETH_BATCH_PAYMENT_BASE + pp.ETH_PER_PAYMENT * 10
-    pp.get_gas_cost_per_payment.return_value = val
-
-    pp.gnt_balance.return_value = gnt * denoms.ether, time.time()
-    pp.eth_balance.return_value = eth * denoms.ether, time.time()
-    pp._gnt_available.return_value = gnt * denoms.ether
-    pp._eth_available.return_value = eth * denoms.ether
-    return pp
-
-
 @patch('signal.signal')
 @patch('golem.network.p2p.node.Node.collect_network_info')
 class TestClientRPCMethods(TestWithDatabase, LogTestCase):
@@ -1019,7 +1020,7 @@ class TestClientRPCMethods(TestWithDatabase, LogTestCase):
         c.resource_server = Mock()
         c.task_server = Mock()
         c.transaction_system.payment_processor = \
-            make_mock_payment_processor(c.transaction_system._sci)
+            make_mock_payment_processor()
 
         task_header = Mock(
             max_price=1 * 10**18,
@@ -1103,7 +1104,7 @@ class TestClientRPCMethods(TestWithDatabase, LogTestCase):
             side_effect=add_task)
 
         c.transaction_system.payment_processor = \
-            make_mock_payment_processor(c.transaction_system._sci)
+            make_mock_payment_processor()
         deferred = c.enqueue_new_task(t_dict)
         task = sync_wait(deferred)
         assert isinstance(task, Task)

